server/interceptor/interceptor.go

```diff
@@ -229,129 +221,51 @@ func isLeafOK(leaf *trillian.QueuedLogLeaf) bool {
 	return leaf == nil || leaf.Status == nil || leaf.Status.Code == int32(codes.OK)
 }
 
-// rpcInfo contains information about an RPC, as extracted from its request message.
 type rpcInfo struct {
-	doNotIntercept bool
+	// getTree and quota enable their corresponding interceptor logic.
+	getTree, quota bool
 
-	// treeID is the tree ID tied to this RPC, if any (zero means no tree).
-	treeID int64
+	readonly bool
+	treeID   int64
+	treeType trillian.TreeType
 
-	// opts is the trees.GetOpts appropriate to this RPC (TreeType, readonly vs readwrite, etc).
-	// opts is not set if doNotIntercept is true.
-	opts trees.GetOpts
-
-	// specs contains the quota specifications for this RPC.
-	specs []quota.Spec
-
-	// tokens is number of quota tokens consumed by this request.
+	specs  []quota.Spec
 	tokens int
 }
 
-// getRPCInfo returns the rpcInfo for the given request, or an error if the request is not mapped.
-// RPCs are mapped using the following logic:
-// treeID is acquired via one of the "Request" interfaces defined below (treeIDRequest,
-// logIDRequest, mapIDRequest, etc). Requests must implement to one of those.
-// TreeType and Readonly are determined based on the request type.
-func getRPCInfo(req interface{}, quotaUser string) (*rpcInfo, error) {
+func newRPCInfo(req interface{}, quotaUser string) (*rpcInfo, error) {
+	// Set "safe" defaults: enable all interception and assume requests are readonly.
+	info := &rpcInfo{getTree: true, quota: true, readonly: true}
 	switch req.(type) {
+
+	// Not intercepted at all
 	case
-		// Do not intercept certain admin requests:
-		// * CreateTree creates new trees (no associated treeID)
-		// * Get, Delete and UndeleteTree may read with deleted trees
-		// * ListTrees returns 0 or more trees
-		*trillian.CreateTreeRequest,
-		*trillian.DeleteTreeRequest,
-		*trillian.GetTreeRequest,
-		*trillian.UndeleteTreeRequest,
-		*trillian.ListTreesRequest,
-		// Do not intercept any quota requests, common rules do not apply.
+		// Quota requests
```

Comment:
`// Quota configuration requests` might help the casual reader  (as opposed to quota-using requests)

Reply:
Done.